src/components/TimeControl/TimeControlPicker.tsx

```diff
@@ -724,3 +754,21 @@ function recallTimeControlSettings(speed: Speed, time_control_system: TimeContro
         ),
     );
 }
+
+function numericInputOnBlur(
+    sender: FocusEvent,
```

Comment:
Also trying to think of a tighter type for `propertyName` since you asked..

Maybe [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads)?

```
function numericInputOnBlur(
    sender: React.FocusEvent<HTMLInputElement, Element>,
    speed: Speed,
    time_control_system: "byoyomi",
    propertyName: "periods",
);
function numericInputOnBlur(
    sender: React.FocusEvent<HTMLInputElement, Element>,
    speed: Speed,
    time_control_system: "canadian",
    propertyName: "stones_per_period",
) {
    ...
}
```

That's probably pretty close to what we want to check for ("periods" if Byoyomi, "stones_per_period" if Canadian), but unfortunately the compiler complains on the callbacks since it thinks `state.system` may have changed by the time the callback is processed.

Reply:
> I think if you change sender to React.FocusEvent<HTMLInputElement, Element>, you won't have to typecast target later

ohh!! i missed the React part... i knew it was possible but couldn't remember it was in React.*...

> That's probably pretty close to what we want to check for ("periods" if Byoyomi, "stones_per_period" if Canadian), but unfortunately the compiler complains on the callbacks since it thinks state.system may have changed by the time the callback is processed.

yes, that's the problem... the compiler can't help much here since the state is dynamic... it will never know if it was called correctly.